sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3083,6 +3083,17 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val LEGACY_DATETIME_ENCODER_ENABLED =
+    buildConf("spark.sql.legacy.datetimeEncoder.enabled")
+      .doc("If the configuration property is set to true, java.sql.Timestamp and " +
+        "java.sql.Date are used as external types for " +
+        "Catalyst's TimestampType and DateType in rows encoding and decoding when the property " +
+        s"${DATETIME_JAVA8API_ENABLED.key} is true (otherwise it is ignored).")
+      .version("3.3.0")
+      .internal()
```

Comment:
For the new config name, we had better remove this `internal` because the parent config is exposed already.

Reply:
The "legacy" configs must be `internal`. Why it is a legacy one is another question.